pkg/expapi/types.go

```diff
@@ -27,3 +27,37 @@ support is experimental.
 */
 
 package expapi
+
+import "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+// ScalerSpec describes the attributes a Scaler subresource
+type ScalerSpec struct {
+	// Replicas is the number of desired replicas. This is a pointer to distinguish between explicit zero and unspecified.
+	Replicas *int `json:"replicas,omitempty" description:"number of replicas desired; defaults to 1; see http://releases.k8s.io/HEAD/docs/user-guide/replication-controller.md#what-is-a-replication-controller"`
+}
+
+// ScalerStatus represents the current status of a Scaler subresource.
+type ScalerStatus struct {
+	// Replicas is the number of actual replicas.
+	Replicas int `json:"replicas" description:"most recently oberved number of replicas; see http://releases.k8s.io/HEAD/docs/user-guide/replication-controller.md#what-is-a-replication-controller"`
+
+	// Selector is a label query over pods that should match the replicas count.
+	Selector map[string]string `json:"selector,omitempty" description:"label keys and values that must match in order to be controlled by this replication controller, if empty defaulted to labels on Pod template; see http://releases.k8s.io/HEAD/docs/user-guide/labels.md#label-selectors"`
```

Comment:
Currently, the selector is for pods (and autoscaler is also only for pods). You don't need to mutate the selector, please note that it is a part of status. The selector will be set by RC/deployment bind to scaler.